RELEASING.md

```diff
@@ -26,10 +33,21 @@ This project does not have a strict release schedule. However, we would make a r
 - Run `ddev release changelog . <NEW_VERSION>` to update the `CHANGELOG.md` file at the root of this repository
 - Commit the changes to the repository in a release branch and get it approved/merged.
 
-## Release
+## Release Process
 
 Our team will trigger the release pipeline.
-- Checkout the [tag release](#commands).
-- Run `ddev release build .` and `ddev release upload --sdist`.
-  - _Prerequisites_: setup PyPi Tooling. 
+
+### Prerequisite
+
+#### Setup PyPI
+
+In order to create a new release you will need to setup PyPI Tooling as well as `datadog_checks_dev` mentioned above.
+See our internal documentation for more details.
+
+### Release
+
+1. Tag the release on GitHub. [Example](https://github.com/DataDog/datadogpy/releases/tag/v0.10.0)
+2. Checkout the [tag release](#commands).
+3. Run `ddev release build .` and `ddev release upload --sdist`.
   - Make sure that both an `sdist` and a [universal wheel](https://packaging.python.org/guides/distributing-packages-using-setuptools/#universal-wheels) have been uploaded to [PyPI](https://pypi.python.org/pypi/datadog/).
+4. Set CHANGELOG release date. [Example](https://github.com/DataDog/datadogpy/commit/e89b19c0bc1e5ea2ec026e14b11d05b980062ffb)
```

Comment:
Can this be part of the changelog creation process? If the release is done in a short timeframe, we could just have one PR 😄